.github/workflows/ci-pipeline.yml

```diff
@@ -15,4 +15,31 @@ jobs:
       - name: Build docker image for tests
         run: docker build --target test --tag todo-app:test .
       - name: Run tests from docker image
-        run: docker run todo-app:test 
\ No newline at end of file
+        run: docker run todo-app:test 
+  
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v2
+      - name: Log into dockerhun
+        run: echo ${{ secrets.DOCKERHUB_ACCESS_TOKEN }} | docker login -u schandarana --password-stdin
+      - name: Build production image
+        run: docker build --target production --tag schandarana/todo-app:$GITHUB_SHA --tag schandarana/todo-app:latest .
+      - name: Push image to dockerhub
+        run: docker push schandarana/todo-app --all-tags
+      - name: Update image tag 
+        run: docker tag schandarana/todo-app:latest registry.heroku.com/shiv-devops-course-starter/web
```

Comment:
Given that you have your manual build step in line 30, could we just attach this tag as part of that step at the same time? Arguably it might be overloading that line a bit too much mind..